internal/driver/driver.go

```diff
@@ -0,0 +1,421 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2022 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package driver
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"sync"
+	"time"
+
+	sdkModel "github.com/edgexfoundry/device-sdk-go/v2/pkg/models"
+	sdk "github.com/edgexfoundry/device-sdk-go/v2/pkg/service"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap/secret"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap/startup"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/config"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/clients/logger"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/errors"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+
+	"github.com/IOTechSystems/onvif"
+	"github.com/IOTechSystems/onvif/device"
+	wsdiscovery "github.com/IOTechSystems/onvif/ws-discovery"
+)
+
+var once sync.Once
+var driver *Driver
+
+const (
+	URLRawQuery = "urlRawQuery"
+	jsonObject  = "jsonObject"
+)
+
+// Driver implements the sdkModel.ProtocolDriver interface for
+// the device service
+type Driver struct {
+	lc            logger.LoggingClient
+	asynchCh      chan<- *sdkModel.AsyncValues
+	deviceCh      chan<- []sdkModel.DiscoveredDevice
+	config        *configuration
+	lock          *sync.RWMutex
+	deviceClients map[string]*DeviceClient
```

Comment:
EdgeX already has a `DeviceClient` be more specific to avoid confusion.
```suggestion
	onvifClients map[string]*OnvifClient
```

Reply:
Renamed.